### PR TITLE
Update development dependencies

### PR DESCRIPTION
**Pull request by watsonbox**

### CODE REVIEW

File: spec/checker_jobs/notifiers/bugsnag_spec.rb
Comment (godric) [inline spec/checker_jobs/notifiers/bugsnag_spec.rb:39]: Did the changes bellow invalid what is said in this comment ?
In either case shouldn't it be reflected in the `gemspec` version requirements ?

**Review by godric — CHANGES_REQUESTED** [review]
It'd be nice to wait until after #191 is merged so we can rebase this and get some CI support 🙏

File: .rubocop.yml
Comment (godric) [inline .rubocop.yml:4]: This is automatic from the `.ruby-version` and should probably be left out.

File: spec/checker_jobs/notifiers/bugsnag_spec.rb
Comment (godric) [reply spec/checker_jobs/notifiers/bugsnag_spec.rb:37]: 👋

**Review by godric — CHANGES_REQUESTED** [review]
I think you missed by last comment, I added a new one about the redundant `TargetRubyVersion`  in `rubocop.yml`

File: spec/checker_jobs/notifiers/bugsnag_spec.rb
Comment (watsonbox) [reply spec/checker_jobs/notifiers/bugsnag_spec.rb:37]: Sorry yes I did miss it. Well, nothing in the runtime code changed between these major versions, it's just the specs. So technically you could work with Sidekiq 5 in development just fine, though the specs would fail.

Since it's only development deps, I decided to do as you suggest and adapt the `gemspec`.

**Comment by watsonbox** [comment]
> I think you missed by last comment, I added a new one about the redundant `TargetRubyVersion` in `rubocop.yml`

Indeed, I updated that and thanks also for the simplifaction of `rubocop.yml`.

**Comment by watsonbox** [comment]
Third time's the charm 🤞

**Review by godric — APPROVED** [review]
LGTM!

**Pull request merged**
